### PR TITLE
Rename Connector to Datalinks

### PR DESCRIPTION
**Pull request by MrFlashAccount**
#### Tl;dr 
Closes: enso-org/cloud-v2#1132
This PR renames Connector to Datalink

---

#### Test Plan:
Everything should work as before,  but instead of sending `connectorId` we send `dataLinkId` and endpoint now is `/datalink`

---

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by somebody1234 — APPROVED** [review]
CR :white_check_mark:, QA not done (it's late here)

File: app/ide-desktop/lib/dashboard/src/layouts/AssetProperties.tsx [hunk 1]
```diff
@@ -273,8 +273,8 @@ export default function AssetProperties(props: AssetPropertiesProps) {
                           const oldDataLinkValue = dataLinkValue
                           try {
                             setDataLinkValue(editedDataLinkValue)
-                            await backend.createConnector({
-                              connectorId: item.item.id,
+                            await backend.createDataLink({
+                              dataLinkId: item.item.id,
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/layouts/AssetProperties.tsx]: its `datalinkId` not `dataLinkId`

File: app/ide-desktop/lib/dashboard/src/services/Backend.ts [hunk 2]
```diff
@@ -655,7 +655,7 @@ export enum AssetType {
   project = 'project',
   file = 'file',
   secret = 'secret',
-  dataLink = 'connector',
+  dataLink = 'dataLink',
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/services/Backend.ts]: same rename to `datalink`

File: app/ide-desktop/lib/dashboard/src/services/Backend.ts [hunk 3]
```diff
@@ -1321,11 +1321,11 @@ export default abstract class Backend {
   /** Return file details. */
   abstract getFileDetails(fileId: FileId, title: string): Promise<FileDetails>
   /** Create a Data Link. */
-  abstract createConnector(body: CreateConnectorRequestBody): Promise<ConnectorInfo>
+  abstract createDataLink(body: CreateDataLinkRequestBody): Promise<DataLinkInfo>
   /** Return a Data Link. */
-  abstract getConnector(connectorId: ConnectorId, title: string | null): Promise<Connector>
+  abstract getDataLink(dataLinkId: DataLinkId, title: string | null): Promise<DataLink>
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/services/Backend.ts]: all occurrences of `dataLink` or `dataLinkId` should be renamed to `datalink` or `datalinkId`

File: app/ide-desktop/lib/dashboard/src/services/remoteBackendPaths.ts [hunk 4]
```diff
@@ -47,8 +47,8 @@ export const UPLOAD_FILE_PATH = 'files'
 export const CREATE_SECRET_PATH = 'secrets'
 /** Relative HTTP path to the "list secrets" endpoint of the Cloud backend API. */
 export const LIST_SECRETS_PATH = 'secrets'
-/** Relative HTTP path to the "create connector" endpoint of the Cloud backend API. */
-export const CREATE_CONNECTOR_PATH = 'connectors'
+/** Relative HTTP path to the "create data link" endpoint of the Cloud backend API. */
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/services/remoteBackendPaths.ts]: `create datalink`

**Review by PabloBuchu — CHANGES_REQUESTED** [review]
its Datalink not Data Link so the camel case format is `datalink`

**Comment by MrFlashAccount** [comment]
@PabloBuchu renamed, please take a look once again.

@somebody1234 I didn't touch the DataLink schema fields, not sure we can safely touch it

**Comment by somebody1234** [comment]
> didn't touch the DataLink schema fields

we don't need to touch the DataLink schema right?
but if it's just the *descriptions* then it should be fine. we definitely don't want to be changing the actual shape otherwise the generated JSON will no longer be valid inputs for libs.

i assume it's safe to rename the variant names as well - libs doesn't appear to depend on the JSON Schema file:
https://github.com/search?q=repo%3Aenso-org%2Fenso+dataLinkSchema&type=code

**Comment by somebody1234** [comment]
you should probably update the path in codeowners though (`dataLink` -> `datalink`):
https://github.com/enso-org/enso/blob/a5ca9f9ac22a6a87ab4f9ec13c1ba4a8cbaaed62/.github/CODEOWNERS#L50

**Comment by MrFlashAccount** [comment]
> you should probably update the path in codeowners though (dataLink -> datalink):

Oh, yeah, thanks for noticing!

**Review by GregoryTravis — APPROVED** [review]

**Review by radeusgd — APPROVED** [review]
We probably need a similar PR renaming the usages in our libs. I was always calling it `Data Link` as I was not aware of any 'official' spelling.

**Review by PabloBuchu — APPROVED** [review]
